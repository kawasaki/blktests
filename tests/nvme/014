#!/bin/bash
# SPDX-License-Identifier: GPL-2.0+
# Copyright (c) 2017-2018 Western Digital Corporation or its affiliates.
#
# Test NVMeOF flush command from host with a block device backed ns.

. tests/nvme/rc

DESCRIPTION="flush a NVMeOF block device-backed ns"
QUICK=1

requires() {
	_nvme_requires
	_have_loop
	_require_nvme_trtype_is_fabrics
}

test() {
	echo "Running ${TEST_NAME}"

	_setup_nvmet

	local nvmedev
	local size
	local bs
	local count

	_nvmet_target_setup

	_nvme_connect_subsys "${nvme_trtype}" "${def_subsysnqn}"

	nvmedev=$(_find_nvme_dev "${def_subsysnqn}")
	_check_uuid "${nvmedev}"

	size="$(blockdev --getsize64 "/dev/${nvmedev}n1")"
	bs="$(blockdev --getbsz "/dev/${nvmedev}n1")"
	count=$((size / bs))

	dd if=/dev/urandom of="/dev/${nvmedev}n1" \
		count="${count}" bs="${bs}" status=none

	nvme flush "/dev/${nvmedev}" -n 1

	_nvme_disconnect_subsys "${def_subsysnqn}"

	_nvmet_target_cleanup

	echo "Test complete"
}
