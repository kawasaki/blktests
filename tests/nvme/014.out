Running nvme/014
NVMe Flush: success
NQN:blktests-subsystem-1 disconnected 1 controller(s)
Test complete
